geonode/harvesting/harvesters/wms.py

```diff
@@ -104,66 +136,179 @@ def check_availability(self, timeout_seconds: typing.Optional[int] = 5) -> bool:
+            'phone_facsimile': get_xpath_value(contact_person, "wms:ContactFacsimileTelephone", nsmap),
+            'address_delivery_point': '',
+            'address_city': get_xpath_value(address, "wms:City", nsmap),
+            'address_administrative_area': get_xpath_value(address, "wms:StateOrProvince", nsmap),
+            'address_postal_code': get_xpath_value(address, "wms:PostCode", nsmap),
+            'address_country': get_xpath_value(address, "wms:Country", nsmap),
+            'address_email': get_xpath_value(contact, "wms:ContactElectronicMailAddress", nsmap),
+        }
 
-    def _get_useful_layers(self) -> typing.List[etree.Element]:
-        get_capabilities_response = self.http_session.get(
-            self.remote_url,
-            params={
-                "service": "WMS",
-                "version": "1.3.0",
-                "request": "GetCapabilities",
-            }
-        )
-        get_capabilities_response.raise_for_status()
-        root = etree.fromstring(get_capabilities_response.content, parser=XML_PARSER)
-        nsmap = _get_nsmap(root.nsmap)
-        useful_layers_elements = []
-        leaf_layers = root.xpath("//wms:Layer[not(.//wms:Layer)]", namespaces=nsmap)
-        for layer_element in leaf_layers:
-            title = layer_element.xpath("wms:Title/text()", namespaces=nsmap)[0]
-            if self.layer_title_filter is not None:
-                if self.layer_title_filter.lower() not in title.lower():
-                    continue
-            useful_layers_elements.append(layer_element)
-        return useful_layers_elements
+    def _layer_element_to_json(self, layer_element: etree.Element) -> dict:
+        """Return json of layer from xml element"""
+        nsmap = _get_nsmap(
+            layer_element.nsmap)
+        name = get_xpath_value(
+            layer_element, "wms:Name", nsmap)
+        title = get_xpath_value(
+            layer_element, "wms:Title", nsmap)
+        abstract = get_xpath_value(
+            layer_element, "wms:Abstract", nsmap)
+        try:
+            keywords = get_xpath_elements(
+                get_xpath_elements(layer_element, "wms:KeywordList", nsmap)[0],
+                "wms:Keyword/text()", nsmap
+            )
+            keywords = [str(keyword) for keyword in keywords]
+        except IndexError:
+            keywords = []
+
+        # legend_url link
+        try:
+            legend_url = get_xpath_elements(
+                get_xpath_elements(
+                    get_xpath_elements(layer_element, "wms:Style", nsmap)[0],
+                    "wms:LegendURL", nsmap
+                )[0],
+                "wms:OnlineResource", nsmap
+            )[0].attrib['{' + layer_element.nsmap['xlink'] + '}href']
+        except (IndexError, KeyError):
+            legend_url = ''
+        params = self._base_wms_parameters
+        wms_url = self.remote_url + '?' + urlencode(params)
```

Comment:
It is not clear to me yet just what should be the value of `wms_url`.

This is something to be discussed together with @afabiani

I think we could probably generate a GetMap URL here, so something like:

```
f"{self.remote_url}?service=WMS&version=1.3.0&request=GetMap&..."
``` 

Alternatively we could just keep the `base_url` and then later build the `GetMap` URL, when it is time to show the layer.

Reply:
hi @ricardogsilva 
i checked the geonode and the wms_url that returned by geonode is like that
which is will be same with geonode, i'm not sure
https://somaliland.gep.kartoza.com/api/layers/118/
`{"extension": "html", "link_type": "OGC:WMS", "mime": "text/html", "name": "OGC WMS: geonode Service", "url": "https://somaliland.gep.kartoza.com/geoserver/ows"}, {"extension": "html", "link_type": "OGC:WFS", "mime": "text/html", "name": "OGC WFS: geonode Service", "url": "https://somaliland.gep.kartoza.com/geoserver/ows"}`